core/src/main/java/org/apache/calcite/avatica/ha/RandomSelectLBStrategy.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.avatica.ha;
+
+import org.apache.calcite.avatica.ConnectionConfig;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Random;
+
+/**
+ * Random Select strategy for client side load balancing.
+ */
+public class RandomSelectLBStrategy implements LBStrategy {
```

Comment:
If you choose a random Avatica server every time, the client will (effectively) be stuck in a retry loop to re-create new Connection and Statements on that Avatica server, to then be bounced to a different Avatica server.

I think this implementation would be overtly harmful should a user ever try to configure it.

Reply:
Thanks @joshelser  for the inputs.

For internal failover retries ( done by driver) we have "lb_connection_failover_retries" to s address this issue. By default we will do only 3 failover retries after connection will be marked as failed. 

Not sure if you are pointing about some scenarios related to external retries( implemented by application).